android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/DefaultRepository.kt

```diff
@@ -1070,14 +1070,20 @@ constructor(
         if (currentPage != null && pageSize != null) {
           val fromIndex = currentPage * pageSize
           val toIndex = (currentPage + 1) * pageSize
-          with(searchResults.subList(fromIndex, min(toIndex, searchResults.size))) {
-            mapResourceToRepositoryResourceData(
-              relatedResourcesConfig = relatedResourcesConfig,
-              configComputedRuleValues = configComputedRuleValues,
-              secondaryResourceConfigs = secondaryResourceConfigs,
-              filterActiveResources = filterActiveResources,
-              baseResourceConfig = baseResourceConfig,
-            )
+          val maxSublistIndex = min(toIndex, searchResults.size)
+
+          if (fromIndex < maxSublistIndex) {
```

Comment:
We can add an edge case unit test for when we only have one item in the searchResults _(if there's none already)_